### ISSUE TITLE
[Bug]: Muellmax is not working anymore

### ISSUE THREAD

**Issue by alsFC** [mampfes/hacs_waste_collection_schedule#2468]
### I Have A Problem With:

A specific source

### What's Your Problem

All Muellmax-related sources seem to not work anymore. Calendars for already configured hubs are not filled anymore and new configurations fail with this message in configuration dialog:

Die Quelle hat eine ungültige Antwort zurückgegeben: "Content line could not be parsed into parts: '<!DOCTYPE html>': <!DOCTYPE html>". Bitte überprüfe die Argumente und versuche es erneut.

![Bildschirmfoto 2024-08-16 um 12 57 06](https://github.com/user-attachments/assets/fae0ccd7-044c-449d-8706-a33b7b0d88df)

I've checked muellmax-sources for Münster, Bochum and Haltern

### Source (if relevant)

muellmax

### Logs

```Shell
2024-08-16 12:35:48.828 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Setting default value for service to Hal
2024-08-16 12:35:48.828 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for service: <class 'inspect._empty'>
2024-08-16 12:35:48.828 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Required: service as default type: str
2024-08-16 12:35:48.828 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_ort_sel: <class 'NoneType'>
2024-08-16 12:35:48.828 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_str_sel: <class 'NoneType'>
2024-08-16 12:35:48.828 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_hnr_sel: <class 'NoneType'>
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Setting suggested value for service to Hal (previously filled in)
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for service: <class 'inspect._empty'>
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Required: service as default type: str
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_ort_sel: <class 'NoneType'>
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Setting suggested value for mm_frm_str_sel to Postweg (previously filled in)
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_str_sel: <class 'NoneType'>
2024-08-16 12:35:55.337 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_hnr_sel: <class 'NoneType'>
2024-08-16 12:36:00.221 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Setting suggested value for service to Hal (previously filled in)
2024-08-16 12:36:00.221 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for service: <class 'inspect._empty'>
2024-08-16 12:36:00.222 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Required: service as default type: str
2024-08-16 12:36:00.222 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_ort_sel: <class 'NoneType'>
2024-08-16 12:36:00.222 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Setting suggested value for mm_frm_str_sel to Postweg (previously filled in)
2024-08-16 12:36:00.222 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_str_sel: <class 'NoneType'>
2024-08-16 12:36:00.222 DEBUG (MainThread) [custom_components.waste_collection_schedule.config_flow] Default for mm_frm_hnr_sel: <class 'NoneType'>
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [x] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by SMSCHAUB** [comment]
same here. Muellmax as source and His as service is not working anymore. Tested on version "master" and "2.01". HA running on 2024.8.1, Supervisor on 2024.08.0.. Logs are showing the same errors as mentioned above

**Comment by 5ila5** [comment]
They seem to require a User-Agent for the web requests now for some regions

fixed in #2471 
This will be part of the next release or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
